source/common/http/http1/codec_impl.cc

```diff
@@ -491,35 +504,45 @@ void ConnectionImpl::dispatch(Buffer::Instance& data) {
   // If an upgrade has been handled and there is body data or early upgrade
   // payload to send on, send it on.
   maybeDirectDispatch(data);
+  return ProtobufUtil::Status();
 }
 
-size_t ConnectionImpl::dispatchSlice(const char* slice, size_t len) {
+ProtobufUtil::StatusOr<ssize_t> ConnectionImpl::dispatchSlice(const char* slice, size_t len) {
   ssize_t rc = http_parser_execute(&parser_, &settings_, slice, len);
   if (HTTP_PARSER_ERRNO(&parser_) != HPE_OK && HTTP_PARSER_ERRNO(&parser_) != HPE_PAUSED) {
     sendProtocolError(Http1ResponseCodeDetails::get().HttpCodecError);
+    if (HTTP_PARSER_ERRNO(&parser_) == HPE_HEADER_OVERFLOW) {
```

Comment:
right now, i can't figure out a way to get http-parser to tell me a callback failed from this error, i can only get back callback errors like HPE_CB_HEADER_FIELD etc